src/Lib/Managers/ModeManager.vala

```diff
@@ -53,6 +53,11 @@ public class Akira.Lib.Managers.ModeManager : Object {
 
     public ModeManager (Akira.Lib.ViewCanvas canvas) {
         Object (view_canvas: canvas);
+
+        // any active mode must be ended when user presses Esc.
+        view_canvas.window.event_bus.request_escape.connect (() => {
```

Comment:
For now we can keep this, but this is not the correct implementation of this. We are already filtering key events, so there should be no reason to listen to the event. I may be wrong though, since the event_bus event may trigger outside of explicit esc calls.